feat(yield): display pools info in widget

shoom3301 opened the pull request:
# Summary

Updated the Yield widget UI corresponding to [Figma](https://www.figma.com/design/iIblVIbidg7oG3SdgNCsde/Vampire-Attack-2024-Sept?node-id=0-1&node-type=canvas&t=VjVYtOKdY6kIcp1Y-0).

<img width="497" alt="image" src="https://github.com/user-attachments/assets/85ebb6cc-6b0d-46d7-9ff4-89ad00363064">

<img width="545" alt="image" src="https://github.com/user-attachments/assets/ed4f0f91-d1e1-4d78-8ba4-a1c1dd849fde">

# To Test

1. UI should fit to Figma
2. "Analytics ↗" link should lead to a pool page if it exists. Balancer and Curve UI don't support links with pool address, so we fallback to the all pools list
3. When buy token is and LP token, then display "Deposit" instead of "Swap" in the botton action button
4. APR label should be green only when it's COW AMM pool and it's APR is higher than opossite pool (what if the opposite pool is not an alternative (contains other tokens)?)

elena-zh commented:
Hey @shoom3301 , great!

Some issues: 
1. Percentage value on the info message should be in bold
![шт ищдв](https://github.com/user-attachments/assets/611260f5-81dd-4950-a6b8-364e671f4205)
2. APR for Uni pool is in green when it is higher
![uniswap apr in green](https://github.com/user-attachments/assets/7b4ab141-2448-427c-b00f-1e3b14480a1c)
3. Could you please check why a regular trade returns error?
![regular token pair](https://github.com/user-attachments/assets/f4f7ff58-bbef-4732-a327-fa706f8d41b1)
![regular](https://github.com/user-attachments/assets/5797d723-ddba-4e06-8792-c9aa03f0fe30)
4. Could you please clarify why a pool APR info is not displayed when sell token is a regular token, and a buy token is CoW AMM?
![no details](https://github.com/user-attachments/assets/1867308d-243d-459c-ab4d-57cd9f933252)
However, it is displayed on a controversial state: 
![apr is displayed](https://github.com/user-attachments/assets/2b0575ae-0ddf-4cb9-9e65-eba8f0cb8055)
5.  Also, why pool details are not displayed to not connected users?
![aprs to not connected](https://github.com/user-attachments/assets/1c4de8c7-8c35-425b-91d9-26a9a10dfbfe)
6. In a mobile view, it is impossible to pick a token from the list. I need to open a pool details, and can pick a token only from there. 
7. (edge case): see how balances and token info may be displayed in a mobile interface: 
![image](https://github.com/user-attachments/assets/a8e36514-5ff6-4aec-bc29-ddf9f4c5f008)
8. If I pick a LP token in the sell field, details and link appear in the Buy field. However, the 2nd token has not been selected, Is this OK? 
![details and link when no 2nd token](https://github.com/user-attachments/assets/9d5f38f8-42c1-45bd-bf2e-078e6c8ed9e8)
9. In general, why details/analytics link show data for a sell token? Shouldn't they show details for a buy token? 
![link](https://github.com/user-attachments/assets/8852c69c-143e-4bbf-9de3-2a5c4bdb7776)
10. (reported in previous PRs): balances are in the loading state to a not connected user
![image](https://github.com/user-attachments/assets/f58bb692-ecc4-4020-9e19-e5d1104dc7d9)
11. some spaces are missing in the tooltip 
![image](https://github.com/user-attachments/assets/8922ab57-9c16-442f-85e3-455a57901017)

12. Just FYI: since it is impossible to trade, I cannot check button rename and other related changes (estimated gain, confirm modal, etc.)

shoom3301 commented:
@elena-zh thank you!
1. Fixed
2. Fixed
3. I can'not reproduce the case. Anyway, I've added "Swaps not supported" state when buy token is ERC-20
4. Fixed, now it should be displayed in both places
5. Fixed
6. Fixed
7. Out of this task scope
8. Added Analytics link to the sell token as well
9. Fixed
10. Fixed
11. FIxed
12. Ok

Besided of the fixes, I've added:
1. Display CoW AMM banner in token selector per account
2. "Boost Your Yield with One-Click Conversion". When there is a LP-token with balance and CoW AMM alternative we display full text, otherwise: "Swap any token into CoW AMM pools to start benefiting from attractive APRs"
3. Don't add cowswap fee in Yield widget and in any widget when one of assets is LP-token

elena-zh commented:
Hey @shoom3301 , gteat!

Some issues: 
1. Connect to account, pick a token pair in the Yield field, disconnect the account --> APRs remain to be displayed. It would be nice to hide them
![discon](https://github.com/user-attachments/assets/0f4fab49-b592-4da7-b6f7-3230586ff59a)
2. I have 2 equal pools on CoW AMM and other source, but CoW AMM APR<=than in another pool. When I press on the Booost button, I don't see that token pair is prefilled in the fields. Just a regular form with WETH in the sell field. Is this an expected behavior?
![weth](https://github.com/user-attachments/assets/14a40fda-1aa0-4862-9072-a5c3753c512d)
3. Connect to a wallet, then change account with no LPs. Wait till the banner appear (25+), open token list and press there on the Boost button --> you will be navigated to the Yield field, but the fields are empty there. I think, at least, WETH token should be displayed in the Sell field. 
![image](https://github.com/user-attachments/assets/a322e7d7-7bf2-4380-bbc0-2deec6ab2552)

4. Question for @MindyCoW : could you please review this text and improve it a bit?  :) 
![mindy](https://github.com/user-attachments/assets/e3bad3b7-d3d1-479a-bbc7-182f8cfaffc4)

MindyCoW commented:
Change the hover text to: 
`Use the Swap tab for trades that don't involve an LP token.`

alfetopito commented on apps/cowswap-frontend/src/modules/yield/pure/TargetPoolPreviewInfo.tsx:
```diff
@@ -0,0 +1,73 @@
+            <InfoTooltip>
+              When you swap (sell) <TokenSymbol token={oppositeToken} />, solvers handle the transaction by purchasing
+              the required tokens, depositing them into the pool, and issuing LP tokens to you in return — all in a
+              gas-less operation.
```
Elena raised this before. The gas-less part doesn't apply to SC wallets

alfetopito commented on apps/cowswap-frontend/src/modules/yield/state/poolsInfoAtom.ts:
```diff
@@ -19,30 +19,26 @@ type PoolInfoState = {
 
 const poolsInfoAtom = atomWithStorage<PoolsInfoState>(
-  'poolsInfoAtom:v0',
+  'poolsInfoAtom:v1',
```
Still unreleased. I would not bump it.

shoom3301 replied:
I will reset them before merging in develop.
I usually up them while breaking changes because Elena do tests and it may causes some problems on her side.

elena-zh reviewed (approved):
Approving the PR with the list of [questions ](https://www.notion.so/cownation/Vampire-attack-development-1188da5f04ca80e0943dd119abe0b131?pvs=4#1198da5f04ca80769ce2c23e45910d83)to discuss later
